src/main/java/com/fasterxml/jackson/core/io/CharBufferReader.java

```diff
@@ -0,0 +1,66 @@
+package com.fasterxml.jackson.core.io;
+
+import java.io.Reader;
+import java.nio.CharBuffer;
+
+public class CharBufferReader extends Reader {
+    private final CharBuffer charBuffer;
+
+    public CharBufferReader(CharBuffer buffer) {
+        this.charBuffer = buffer.duplicate();
+    }
+
+    @Override
+    public int read(char[] chars, int off, int len) {
+        int remaining = this.charBuffer.remaining();
+        if (remaining <= 0) {
+            return -1;
+        }
+        int length = Math.min(len, remaining);
+        this.charBuffer.get(chars, off, length);
+        return length;
+    }
+
+    @Override
+    public int read() {
+        if (this.charBuffer.hasRemaining()) {
+            return this.charBuffer.get();
+        }
+        return -1;
+    }
+
+    @Override
+    public long skip(long n) {
+        if (n < 0L) {
+            throw new IllegalArgumentException("number of characters to skip cannot be negative");
+        }
+        int skipped = Math.min((int) n, this.charBuffer.remaining());
```

Comment:
please don't cast long to int - there is an edge case where n is too big and the cast value could be nonsense

Reply:
will change this to:
```suggestion
        int skipped = Math.min(Math.toIntExact(n), this.charBuffer.remaining());
```

Reply:
My suggestion works for values of n greater than max-int. Your suggestion will raise an exception. If the API defines  n to be long, I think we should respect the API and handle large longs.

Reply:
It is a funny set of APIs, `CharBuffer.remaining` returns an int so that suggests you can't have more the max-int chars. So maybe, your suggestion works fine. The skip API accepts longs but it appears that it not something we should worry about.